File: native/bindings/content.cpp
#include "content.h"

namespace cairodart
{

namespace bindings
{

Content::Content(cairo_content_t content) : c(content)
{
}

Content::~Content()
{
}

cairo_content_t Content::content() const
{
    return c;
}

} // bindings

} // cairodart

File: cairodart/lib/src/bindings/content.dart
part of cairodart.base;

abstract class Content extends NativeFieldWrapperClass2 {
  static Content COLOR = new _Content(0x1000);
  static Content ALPHA = new _Content(0x2000);
  static Content COLOR_ALPHA = new _Content(0x3000);
  
  int get value;
}

class _Content extends Content {
  
  final int _value;
  
  _Content(this._value);
  
  int get value => _value;
 
  @override
  operator==(Content other) => value == other.value;
}